### PR TITLE
Guided Onboarding Flow: Store additional data after survey is completed

### PR DESCRIPTION
**Pull request by agrullon95**
Related to https://github.com/Automattic/dotcom-forge/issues/7015

## Proposed Changes

Stores additional data after the Guided onboarding flow (`/start/guided/`) segmentation survey is completed.

`is_dev_account` - Enable developer features for users that select `Creating a site for a client `option for Q1.
`site_intent` - Set site intent to `host-site` for users that select `Creating a site for a client` option for Q1.
`site_goals` - Store site goals
`onboarding_segment` - Store segment based on responses to Q1 and Q2

CleanShot 2024-05-31 at 11 00 37@2x

## Why are these changes being made?

* This data is stored to analyze users' site-building trends and for potential future use cases.

## Testing Instructions

* Checkout PR
* Apply patch: D150692-code
* Navigate to `/start/guided/` and follow the table above to complete the flow.
* Verify the correct `onboarding_segment` is set (values can be seen here: https://github.com/Automattic/wp-calypso/pull/91348/files#diff-41bf9077d219cf1aef332e3b18f974f9ab3e589504f441fe831b45f23ed7c554R55)
* Verify the correct `site_goals` are set ( values can be seen here: )
* Verify the correct `site_intent` is set for `Creating a site for a client` Q1 option (should be set to `host-site`).
* Check `/me/developer` to ensure the developer features option is toggled

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/lib/signup/step-actions/index.js [hunk 1]
```diff
@@ -188,6 +196,10 @@ export const getNewSiteParams = ( {
 			timezone_string: guessTimezone(),
 			wpcom_public_coming_soon: launchAsComingSoon,
 			...( siteAccentColor && { site_accent_color: siteAccentColor } ),
+			...( flowToCheck === 'guided' && setDevAccount && { set_dev_account: setDevAccount } ),
+			...( flowToCheck === 'guided' && siteGoals && { site_goals: siteGoals } ),
+			...( flowToCheck === 'guided' &&
+				onboardingSegment && { onboarding_segment: onboardingSegment } ),
```
Comment (alshakero) [inline client/lib/signup/step-actions/index.js]: Let's make an object called `guidedFlowProps` and populate if flow is guided. 

```js
const guidedFlowProps = {};
if ( flowToCheck === 'guided' ) {
	guidedFlowProps.set_dev_account = setDevAccount;
    .
    .
}
```

Comment (agrullon95) [reply client/lib/signup/step-actions/index.js]: Updated

File: client/lib/signup/step-actions/index.js [hunk 2]
```diff
@@ -156,6 +156,20 @@ export const getNewSiteParams = ( {
 	const siteTitle = ( signupDependencies?.siteTitle || '' ).trim();
 	const selectedDesign = get( signupDependencies, 'selectedDesign', false );
 
+	// Guided onboarding flow specific dependencies
+	const setDevAccount =
+		signupDependencies?.segmentationSurveyAnswers?.[ 'what-brings-you-to-wordpress' ].includes(
+			'client'
+		);
+	const siteGoals = signupDependencies?.segmentationSurveyAnswers?.[ 'what-are-your-goals' ];
+	const onboardingSegment = signupDependencies?.onboardingSegment;
```
Comment (alshakero) [inline client/lib/signup/step-actions/index.js]: Let's move all of these inside the `if`. No need to calculate them otherwise.

**Review by alshakero — APPROVED** [review]
One last comment.

**Pull request merged**